### PR TITLE
Lightweight dark mode

### PR DESCRIPTION
**Pull request by anthonyringoet**
Added a lightweight dark mode. Can certainly be improved, only used some CSS overrides.

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by anthonyringoet** [comment]
Linking to #79

**Comment by anthonyringoet** [comment]
Screenshot_2020-05-04 Fastify, Fast and low overhead web framework, for Node js

File: src/website/content/css/style.scss [hunk 1]
```diff
@@ -269,3 +275,89 @@ a > code {
 .media.v-center {
   align-items: center;
 }
+
+// lightweight dark mode using some overrides
+@media (prefers-color-scheme: dark) {
```
Comment (StarpTech) [inline src/website/content/css/style.scss:301]: I'd prefer a solution with css variables

```css
@media (prefers-color-scheme: dark) {
	:root {
	  --main-bg-color: dark;
	}

}
```

**Review by StarpTech — CHANGES_REQUESTED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

File: src/website/content/css/style.scss (hunk 1 above)
Comment (anthonyringoet) [reply src/website/content/css/style.scss:301]: This is possible for the custom CSS like overriding body background. To override the third-party CSS (eg. Bulma panel and card) you'll still have to overwrite the imported rules afaik. Bulma supports customization but only at compile time with SASS variables afaik.

Is that what you're aiming for here?

Comment (StarpTech) [reply src/website/content/css/style.scss:301]: I agree but did you try this

```scss
// create css variables
$black_value:        hsl(0, 0%, 4%) !default;
// assign css property to bulma variables
$black:         var(--color-black);


// Framework code will work with css property notation
div {
  background-color: #{$black};
}
// Framework code end

// Create css variables
:root {
  --color-black: #{$black_value};
}
```

produces

```scss
div {
  background-color: var(--color-black);
}

:root {
  --color-black: #0a0a0a;
}
```

this should work as long as no other transform functions are applied on `$black`

**Comment by zekth** [comment]
@anthonyringoet could you rebase it and land this? Also we might want a switch for light/dark theme in the future

**Comment by anthonyringoet** [comment]
@zekth rebased the changes from upstream, feel free to check again

**Review by zekth — APPROVED** [review]

**Comment by zekth** [comment]
@StarpTech can you recheck it?

**Comment by anthonyringoet** [comment]
@StarpTech @zekth No need to waste your time rechecking it. FYI, only had time to rebase and check these incoming changes. Did not have the time to add the requested changes.

**Comment by Eomm** [comment]
Closing for https://github.com/fastify/website-next

**Pull request closed without merging**